unlock-app/src/components/interface/locks/metadata/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { useQuery, useMutation } from '@tanstack/react-query'
+import { Button } from '@unlock-protocol/ui'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { useStorageService } from '~/utils/withStorageService'
+import { LockAdvancedForm } from './LockAdvancedForm'
+import { LockCustomForm } from './custom'
+import { LockDetailForm } from './LockDetailForm'
+import { LockTicketForm } from './LockTicketForm'
+import { Attribute, Metadata, MetadataFormData, toFormData } from './utils'
+import { config } from '~/config/app'
+
+export function UpdateLockMetadata() {
+  const router = useRouter()
+  const lockAddress = router.query.address!.toString()
+  const network = Number(router.query.network)
+  const storageService = useStorageService()
+
+  const { data } = useQuery<Record<string, any>>(
+    ['lockMetadata', lockAddress, network],
+    async () => {
+      const response = await storageService.locksmith.lockMetadata(
+        network,
+        lockAddress
+      )
+      return response.data
+    },
+    {
+      onError(error) {
+        console.error(error)
+      },
+      refetchInterval: Infinity,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+      retry: 2,
+    }
+  )
+
+  const methods = useForm<MetadataFormData>({
+    defaultValues: {
+      name: 'Locksmith 101',
+      external_url: 'https://example.com',
```

Comment:
Ideally we'd use a placeholder, rather than set the value so that if the user is not changing it we don't actually link to example.com 👯‍♀️